lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -547,6 +613,47 @@ mod tests {
         assert!(read_segment_1.has_point(12.into()));
     }
 
+    #[test]
+    fn test_points_deduplication() {
+        let dir = Builder::new().prefix("segment_dir").tempdir().unwrap();
+
+        let mut segment1 = build_segment_1(dir.path());
+        let mut segment2 = build_segment_1(dir.path());
+
+        segment1
+            .set_payload(100, 1.into(), &json!({}).into())
+            .unwrap();
+        segment1
+            .set_payload(100, 2.into(), &json!({}).into())
+            .unwrap();
+
+        segment2
+            .set_payload(200, 4.into(), &json!({}).into())
+            .unwrap();
+        segment2
+            .set_payload(200, 5.into(), &json!({}).into())
+            .unwrap();
+
+        let mut holder = SegmentHolder::default();
+
+        let sid1 = holder.add(segment1);
+        let sid2 = holder.add(segment2);
+
+        let res = holder.deduplicate_points().unwrap();
```

Comment:
Maybe we could have a second call to `deduplicate_points` at the end asserting that it found zero points that time.

Reply:
didn't get that

Reply:
I am proposing to call `deduplicate_points` a second time to assert that the function has a side effect.
It should return 0.

Reply:
for debug builds only?

Reply:
Just in this test.
This was a nitpick, I approved the PR :+1: